llvm/include/llvm-c/Visibility.h

```diff
@@ -0,0 +1,44 @@
+/*===-- llvm-c/Visibility.h - Visibility macros for llvm-c ------*- C++ -*-===*\
+|*                                                                            *|
+|* Part of the LLVM Project, under the Apache License v2.0 with LLVM          *|
+|* Exceptions.                                                                *|
+|* See https://llvm.org/LICENSE.txt for license information.                  *|
+|* SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception                    *|
+|*                                                                            *|
+|*===----------------------------------------------------------------------===*|
+|*                                                                            *|
+|* This header defines visibility macros used for the LLVM C interface. These *|
+|* macros are used to annotate C functions that should be exported as part of *|
+|* a shared library or DLL.                                                   *|
+|*                                                                            *|
+\*===----------------------------------------------------------------------===*/
+
+#ifndef LLVM_C_VISIBILITY_H
+#define LLVM_C_VISIBILITY_H
+
+#include "llvm/Config/llvm-config.h"
+
+/// LLVM_C_ABI is the export/visibility macro used to mark symbols declared in
+/// llvm-c as exported when built as a shared library.
+
+#if !defined(LLVM_ABI_GENERATING_ANNOTATIONS)
+// TODO(https://github.com/llvm/llvm-project/issues/145406): eliminate need for
+// two preprocessor definitions to gate LLVM_ABI macro definitions.
+#if defined(LLVM_ENABLE_LLVM_C_EXPORT_ANNOTATIONS) &&                          \
+    !defined(LLVM_BUILD_STATIC)
```

Comment:
We should be able to build llvm-c statically or dynamically separately from LLVM. I think that if we want to guarantee that they are built the same, we should control that at the CMake level.